src/lib.rs

```diff
@@ -284,25 +387,55 @@ pub struct Bus<T> {
     // are waiting
     #[cfg_attr(feature = "cargo-clippy", allow(type_complexity))]
     waiting: (
-        mpsc::Sender<(thread::Thread, usize)>,
-        mpsc::Receiver<(thread::Thread, usize)>,
+        mpsc::Sender<(P, usize)>,
+        mpsc::Receiver<(P, usize)>,
     ),
 
     // channel used to communicate to unparker that a given thread should be woken up
-    unpark: mpsc::Sender<thread::Thread>,
+    unpark: mpsc::Sender<P>,
 
     // cache used to keep track of threads waiting for next write.
     // this is only here to avoid allocating one on every broadcast()
-    cache: Vec<(thread::Thread, usize)>,
+    cache: Vec<(P, usize)>,
+}
+
+impl<T> Bus<T, thread::Thread> {
+    /// TODO forwards to with_parkable
+    pub fn new(len: usize) -> Self {
+        Bus::with_parkable(len)
+    }
+
+    /// Broadcasts a value on the bus to all consumers.
+    ///
+    /// This function will block until space in the internal buffer becomes available.
+    ///
+    /// Note that a successful send does not guarantee that the receiver will ever see the data if
+    /// there is a buffer on this channel. Items may be enqueued in the internal buffer for the
+    /// receiver to receive at a later time. Furthermore, in contrast to regular channels, a bus is
+    /// *not* considered closed if there are no consumers, and thus broadcasts will continue to
+    /// succeed.
+    pub fn broadcast(&mut self, val: T) {
+        if let Err(..) = self.broadcast_inner(val, <thread::Thread as Parkable>::prepare(BlockCondition::Block)) {
+            unreachable!("blocking broadcast_inner can't fail");
+        }
+    }
 }
 
-impl<T> Bus<T> {
+#[cfg(feature = "async")]
+impl<T> Bus<T, futures::task::Task> {
+    /// TODO forwards to with_parkable
+    pub fn new_async(len: usize) -> Self {
```

Comment:
There really isn't a reason for this to have a different name than `Bus<T, Thread>::new`. In most cases, type inference can determine which one should be used (for example, populating a struct field), and other examples can be resolved by specifying the type parameter at construction.

Reply:
Given also that this is equivalent to `new`, it may even make sense to get rid of these, rename `with_parkable` to `new` and make it public (it seems to be already, though it was intended to be some kind of implementation detail).